src/DictVectors/abstractdvec.jl

```diff
@@ -56,225 +58,148 @@ end
-"""
-struct NormProjector <: AbstractProjector end
-
-LinearAlgebra.dot(::NormProjector, y::DVecOrVec) = norm(y, 1)
 
-"""
-    Norm2Projector() <: AbstractProjector
-Results in computing the two-norm when used in `dot()`. E.g.
-```julia
-dot(NormProjector(),x)
--> norm(x,2) # with type Float64
-```
-
-See also [`PostStepStrategy`](@ref Main.PostStepStrategy), and [`AbstractProjector`](@ref) for use
-of projectors in [`lomc!`](@ref Main.lomc!).
-"""
-struct Norm2Projector <: AbstractProjector end
-
-LinearAlgebra.dot(::Norm2Projector, y::DVecOrVec) = norm(y, 2)
-# NOTE that this returns a `Float64` opposite to the convention for
-# dot to return the promote_type of the arguments.
+LinearAlgebra.dot(v::AbstractDVec, w::AbstractDVec) = inner(v, w)
 
-"""
-    Norm1ProjectorPPop() <: AbstractProjector
-Results in computing the one-norm per population when used in `dot()`. E.g.
-```julia
-dot(Norm1ProjectorPPop(),x)
--> norm(real.(x),1) + im*norm(imag.(x),1)
-```
-
-See also [`PostStepStrategy`](@ref Main.PostStepStrategy), and [`AbstractProjector`](@ref) for use
-of projectors in [`lomc!`](@ref Main.lomc!).
-"""
-struct Norm1ProjectorPPop <: AbstractProjector end
-
-function LinearAlgebra.dot(::Norm1ProjectorPPop, y::DVecOrVec)
-    T = float(valtype(y))
-    if T <: Complex
-        return T(sum(values(y)) do p
-            abs(real(p)) + im*abs(imag(p))
-        end)
+function LinearAlgebra.norm(v::AbstractDVec, p::Real=2)
+    T = float(promote_type(valtype(v), typeof(p)))
+    if p === 1
+        return sum(abs, values(v); init=zero(T))
+    elseif p === 2
+        return sqrt(sum(abs2, values(v); init=zero(T)))
+    elseif p === Inf
+        return mapreduce(abs, max, values(v), init=real(zero(T)))
     else
-        return dot(NormProjector(), y)
+        error("$p-norm of $(typeof(v)) is not implemented.")
     end
 end
 
-# NOTE that this returns a `Float64` opposite to the convention for
-# dot to return the promote_type of the arguments.
-# NOTE: This operation should work for `MPIData` and is MPI synchronizing
-
-"""
-    PopsProjector() <: AbstractProjector
-Results in computing the projection of one population on the other
-when used in `dot()`. E.g.
-```julia
-dot(PopsProjector(),x)
--> real(x) ⋅ imag(x)
-```
-
-See also [`PostStepStrategy`](@ref Main.PostStepStrategy), and [`AbstractProjector`](@ref) for use
-of projectors in [`lomc!`](@ref Main.lomc!).
-"""
-struct PopsProjector <: AbstractProjector end
-
-function LinearAlgebra.dot(::PopsProjector, y::DVecOrVec)
-    T = float(real(valtype(y)))
-    return T(sum(values(y)) do p
-        real(p) * imag(p)
-    end)
-end
+LinearAlgebra.normalize!(v::AbstractDVec, p::Real=2) = scale!(v, inv(norm(v, p)))
+LinearAlgebra.normalize(v::AbstractDVec, p::Real=2) = normalize!(copy(v), p)
 
 """
-    walkernumber(w)
+    walkernumber(v)
 
 Compute the number of walkers in `w`. It is used for updating the shift. Overload this
 function for modifying population control.
```

Comment:
```suggestion
    walkernumber(v)

Compute the number of walkers in `v`. It is used for updating the shift. Overload this
function for modifying population control.
```
Also needs to be changed in the following line (but github won't let me).